tests/ServiceCollectionExtensionsShould.cs

```diff
@@ -11,28 +14,208 @@ public class ServiceCollectionExtensionsShould
+        {
+            var services = new ServiceCollection();
+
+            services.AddKorm("server=localhost-1", "db1");
+            services.AddKorm("server=localhost-2");
+            services.AddKorm("server=localhost-3", "db3");
+
+            ServiceProvider provider = services.BuildServiceProvider();
+            IDatabaseFactory factory1 = provider.GetService<IDatabaseFactory>();
+            IDatabase db1 = factory1.GetDatabase("db1");
+            IDatabase db2 = factory1.GetDatabase("db1");
+            IDatabase db3 = null;
+            using (IServiceScope scope = provider.CreateScope())
+            {
+                IDatabaseFactory factory2 = scope.ServiceProvider.GetService<IDatabaseFactory>();
+                db3 = factory2.GetDatabase("db1");
+            }
+
+            db2.Should().Be(db1, "\"db2\" was created using the same factory as \"db1\".");
+            db3.Should().NotBeNull().And.NotBe(db1, "\"db3\" was created using factory in different scope as \"db1\".");
+        }
+
+        [Fact]
+        public void ReturnDatabaseWithDefaultName()
+        {
+            var services = new ServiceCollection();
+            DatabaseConfigurationBase config1 = Substitute.For<DatabaseConfigurationBase>();
+            DatabaseConfigurationBase config2 = Substitute.For<DatabaseConfigurationBase>();
+            services.AddKorm("server=localhost-1", "db1").UseDatabaseConfiguration(config1);
+            services.AddKorm("server=localhost-2").UseDatabaseConfiguration(config2);
+
+            ServiceProvider provider = services.BuildServiceProvider();
+            IDatabaseFactory factory = provider.GetService<IDatabaseFactory>();
+
+            _ = factory.GetDatabase();
+            config2.Received().OnModelCreating(Arg.Any<ModelConfigurationBuilder>());
+            config1.DidNotReceive().OnModelCreating(Arg.Any<ModelConfigurationBuilder>());
+        }
+
+        [Fact]
+        public void ThrowExceptionWhenAddingDatabaseWithTheSameName()
+        {
+            var services = new ServiceCollection();
+            services.AddKorm("server=localhost-1", "db1");
+
+            Action action = () => services.AddKorm("server=localhost-2", "db1");
+
+            action.Should().Throw<ArgumentException>().And.ParamName.Should().Be("name");
+        }
+
+        [Fact]
+        public void ThrowExceptionWhenPassingInvalidNameToDatabaseFactory()
+        {
+            var services = new ServiceCollection();
+            services.AddKorm("server=localhost-1", "db1");
+            services.AddKorm("server=localhost-2", "db2");
+            ServiceProvider provider = services.BuildServiceProvider();
+            IDatabaseFactory factory = provider.GetService<IDatabaseFactory>();
+
+            Action action = () => factory.GetDatabase("NonExistingName");
+
+            action.Should().Throw<ArgumentException>().And.ParamName.Should().Be("name");
+        }
+
+        [Fact]
```

Comment:
I would add one test to verify that `IDatabase` instance is disposed after the scope ended.